synapse/lib/layer.py

```diff
@@ -4093,6 +4094,71 @@ async def iterNodeDataKeys(self, buid):
             prop = self.getAbrvProp(abrv)
             yield prop[0]
 
+    async def iterLayerAddPerms(self):
+
+        # nodes & props
+        for byts, abrv in self.propabrv.slab.scanByFull(db=self.propabrv.name2abrv):
+            form, prop = s_msgpack.un(byts)
+            if form is None or prop is None:
```

Comment:
prop being None is when you should check the `('node', 'add', form)` perm. For the various checks you should also do a `scanByPref` using the abrv on the corresponding db to ensure there is at least 1 value still present (if they set one prop and deleted it, the abrv would still be present but they wouldn't need that perm anymore)